### DISCUSSION TITLE
[Chapter 12] @Async 와 스레드풀 설정

### DISCUSSION THREAD

**Discussion by JoisFe** [cowaine/spring-boot-for-msa#190] [Chapter 12. Spring Events] [answered]
@Async를 사용하여 비동기 이벤트처리를 할때 책에서는 스레드 풀을 설정하고 있습니다.
만약 스레드 풀을 설정하면 매번 스레드 생성, 삭제를 하며 발생하는 오버헤드를 줄일 수 있는 장점이 있습니다.
그런데 만약 스레드 풀을 설정하지 않는다면 (max pool size 를 지정하지 않는다면) 수많은 비동기 이벤트가 발생했을 경우 제한 없이 스레드가 생성되는 것인지 궁금합니다.

그리고 스레드풀 설정시 core pool size, max pool size 등을 어만큼 설정해야할지 고민이 많은데 어떻게 해당 값을 설정하는지 기준이 있으시다면 공유해주시면 감사하겠습니다.

**Comment by corock** [comment, marked as answer]
> 그런데 만약 스레드 풀을 설정하지 않는다면 (max pool size 를 지정하지 않는다면) 수많은 비동기 이벤트가 발생했을 경우 제한 없이 스레드가 생성되는 것인지 궁금합니다.

이런 경우 필요할 때마다 스레드를 생성하는 작업이 실행될 것 같습니다.
트래픽이 많아지면 이 행위 자체가 부하를 줄 수 있겠네요.

> 그리고 스레드풀 설정시 core pool size, max pool size 등을 어만큼 설정해야할지 고민이 많은데 어떻게 해당 값을 설정하는지 기준이 있으시다면 공유해주시면 감사하겠습니다.

저도 요즘 이 수치의 중요성을 느끼고 있고, 이에 대해서는 인스턴스의 사양과 운영 중인 대수 등에 따라 적정 수치가 달라져야 할 것 같다는 의견 드립니다.

**Reply by corock** [reply]
_cf._ [HikariCP Dead lock에서 벗어나기 (실전편)](https://techblog.woowahan.com/2663)
